ProptechOS-Edge-Interface/examples/java/src/main/java/com/proptechos/service/observations/TelemetryGenerator.java

```diff
@@ -18,33 +20,77 @@
   private Random rand;
 
   private static final int MIN_TEMPERATURE = 15;
+  private static final int MIN_HUMIDITY = 40;
+  private static final int MIN_CO2 = 700;
+
+  private static final String TEMPERATURE_SENSOR_QUANTITY_KIND = "Temperature";
+  private static final String HUMIDITY_SENSOR_QUANTITY_KIND = "Humidity";
+  private static final String CO2_SENSOR_QUANTITY_KIND = "CO2";
 
   public TelemetryGenerator(DeviceConfig deviceConfig) {
     this.deviceConfig = deviceConfig;
     this.rand = new Random();
   }
 
   public RecMessage generateTelemetry() {
-    final Sensor randomSensor = getRandomSensor();
+    final Sensor temperatureSensor = getTemperatureSensor();
+    final Sensor humiditySensor = getHumiditySensor();
+    final Sensor co2Sensor = getCO2Sensor();
     //we use temperature from actuation, in case
     //when there was no actuation from ProptechOS we use some randomly generated one
     Object temperatureValue = isNull(actuationValue)
         ? generateRandomTemperature()
         : actuationValue;
-    final RecObservation observation = new RecObservation(ZonedDateTime.now(ZoneOffset.UTC),
-        temperatureValue, randomSensor.getQuantityKind(), randomSensor.getSensorId());
+    // for humidity we use randomly generated values
+    Object humidityValue = generateRandomHumidity();
+    // for CO2 we use randomly generated values
+    Object co2Value = generateRandomCO2();
+    final RecObservation temperetureObservation = new RecObservation(ZonedDateTime.now(ZoneOffset.UTC),
+        temperatureValue, temperatureSensor.getQuantityKind(), temperatureSensor.getSensorId());
+    final RecObservation humidityObservation = new RecObservation(ZonedDateTime.now(ZoneOffset.UTC),
+            humidityValue, humiditySensor.getQuantityKind(), humiditySensor.getSensorId());
+    final RecObservation co2Observation = new RecObservation(ZonedDateTime.now(ZoneOffset.UTC),
+            co2Value, co2Sensor.getQuantityKind(), co2Sensor.getSensorId());
+    List<RecObservation> observations = new ArrayList<>();
+    observations.add(temperetureObservation);
+    observations.add(humidityObservation);
+    observations.add(co2Observation);
```

Comment:
move this to separate method.
also create method that creates observations, and generalize it as much as possible(use method reference).
and use Arrays.asList() to create list with observation